catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/AbstractAssociatedMetacardsHandler.java

```diff
@@ -0,0 +1,151 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformer.impl;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractAssociatedMetacardsHandler {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(AbstractAssociatedMetacardsHandler.class);
+
+  protected CatalogFramework catalogFramework;
+
+  protected EndpointUtil util;
+
+  protected FilterBuilder filterBuilder;
+
+  public AbstractAssociatedMetacardsHandler(
+      CatalogFramework catalogFramework, EndpointUtil util, FilterBuilder filterBuilder) {
+    this.catalogFramework = catalogFramework;
+    this.util = util;
+    this.filterBuilder = filterBuilder;
+  }
+
+  public void handle(Metacard existing, Metacard updated, List<Metacard> queries) {
+    create(existing, updated, queries);
+    delete(existing, updated, queries);
+    update(existing, updated, queries);
+  }
+
+  abstract void create(Metacard existing, Metacard updated, List<Metacard> metacards);
+
+  abstract void update(Metacard existing, Metacard updated, List<Metacard> metacards);
+
+  abstract void delete(Metacard existing, Metacard updated, List<Metacard> metacards);
+
+  List<Serializable> getMetacardIds(List<Metacard> metacards) {
+    return metacards.stream().map(Metacard::getId).collect(Collectors.toList());
+  }
+
+  List<Serializable> getAssociatedMetacardIds(Metacard workspace, String association) {
+    Attribute attr = workspace.getAttribute(association);
+    return attr == null ? new ArrayList<>() : attr.getValues();
+  }
+
+  boolean isAssociated(Metacard workspace, Metacard metacard, String association) {
+    return getAssociatedMetacardIds(workspace, association).contains(metacard.getId());
+  }
+
+  Metacard getMetacard(String id) {
+    try {
+      return util.getMetacard(id);
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      LOGGER.debug("Failed to retrieve query metacard {}", id, e);
+    }
+
+    return null;
+  }
+
+  List<Metacard> createMetacard(List<Metacard> metacards) {
```

Comment:
Should be named `createMetacards`, `updateMetacards`, and `deleteMetacards`, respectively.